src/proto/validation.proto

```diff
@@ -0,0 +1,41 @@
+syntax = "proto2";
+
+package TbotsProto;
+
+import "proto/game_state.proto";
+import "proto/team.proto";
+import "proto/vision.proto";
+import "proto/geometry.proto";
+import "proto/ball.proto";
+
+enum ValidationStatus
+{
+    PASSING = 0;
+    FAILING = 1;
+}
+
+enum ValidationType
+{
+    EVENTUALLY = 0;
+    ALWAYS     = 1;
+}
+
+message ValidationGeometry
+{
+    repeated Polygon polygons = 1;
+    repeated Circle circles   = 2;
+    repeated Vector vectors   = 3;
+}
+
+message ValidationProto
+{
+    required string failure_msg             = 1;
```

Comment:
shouldn't `failure_msg` not be `required`, since it is only applicable when `validation_status` is `FAILING`?

Reply:
Its the msg to print when there is a failure so its always populated